.travis.yml

```diff
@@ -0,0 +1,103 @@
+    - OPENSSL_CFLAGS="-I$OPENSSL_PREFIX/include"
+    - OPENSSL_LIBS="-L$OPENSSL_PREFIX/lib -lssl -lcrypto"
+    - LD_LIBRARY_PATH=~/opt/ssl/lib:$LD_LIBRARY_PATH
+
+matrix:
+  include:
+    - env: CONFIG="--with-crypto-library=openssl"
+      compiler: gcc
+    - env: CONFIG="--with-crypto-library=openssl"
+      compiler: clang
+    - env: CONFIG="--with-crypto-library=openssl --disable-lzo"
+      compiler: gcc
+    - env: CONFIG="--with-crypto-library=openssl --disable-lzo"
+      compiler: clang
+    - env: CONFIG="--with-crypto-library=mbedtls"
+      compiler: gcc
+    - env: CONFIG="--with-crypto-library=mbedtls"
+      compiler: clang
+    - env: CONFIG="--with-crypto-library=openssl"
+      os: osx
+      osx_image: xcode6.4
+      compiler: clang
+    - env: CONFIG="--with-crypto-library=mbedtls"
+      os: osx
+      osx_image: xcode7.3
+      compiler: clang
+    - env: CONFIG="--with-crypto-library=mbedtls"
+      os: osx
+      osx_image: xcode6.4
+      compiler: clang
+  allow_failures:
+    # These currently do not work on OS X but are being worked on by @chipitsine
+    - env: CONFIG="--with-crypto-library=openssl"
+      os: osx
+      osx_image: xcode6.4
+      compiler: clang
+    - env: CONFIG="--with-crypto-library=mbedtls"
+      os: osx
+      osx_image: xcode6.4
+      compiler: clang
+  exclude:
+    - compiler: gcc
+
+addons:
+  apt:
+    packages:
+      - liblzo2-dev
+      - libpam0g-dev
+      - liblz4-dev
+      - linux-libc-dev
+
+cache:
+  apt: true
+  ccache: true
+  directories:
+  - download-cache
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update     ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install lzo; fi
+
+install:
+  - mkdir ~/bin && ln -s $(which ccache) ~/bin/$CC && export PATH=$HOME/bin:$PATH
+  - if [ ! -d download-cache ]; then mkdir download-cache; fi
+  - if [ ! -f download-cache/mbedtls-${MBEDTLS_VERSION}-apache.tgz ]; then wget -O download-cache/mbedtls-${MBEDTLS_VERSION}-apache.tgz https://tls.mbed.org/download/mbedtls-${MBEDTLS_VERSION}-apache.tgz; fi
+  - if [ ! -f download-cache/openssl-${OPENSSL_VERSION}.tar.gz ]; then wget -O download-cache/openssl-${OPENSSL_VERSION}.tar.gz https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz; fi
+  - tar zxf download-cache/mbedtls-${MBEDTLS_VERSION}-apache.tgz
+  - cd mbedtls-${MBEDTLS_VERSION} && make > build.log 2>&1 || (cat build.log && exit 1)
+  - make install DESTDIR=$MBEDTLS_PREFIX && cd ..
+  - tar zxf download-cache/openssl-${OPENSSL_VERSION}.tar.gz
+  - cd openssl-${OPENSSL_VERSION}/
+  - ./config shared --prefix=$OPENSSL_PREFIX -DPURIFY > build.log 2>&1 || (cat build.log && exit 1)
+  - make -j$JOBS > build.log 2>&1 || (cat build.log && exit 1)
+  - make install_sw > build.log 2>&1 || (cat build.log && exit 1)
+  - cd ..
+
+script:
+  - autoreconf -vi
+  - ./configure $CONFIG || (cat config.log && exit 1)
+  - make -j$JOBS > build.log 2>&1 || (cat build.log && exit 1)
```

Comment:
The `command > build.log 2>&1  || cat build.log && exit 1` approach should work just fine. Everything is redirected to build.log, but if (and only if) the command return a non-zero exit code, then the entire output pushed to STDOUT.

Now the question is whether we want to see output from non-failing builds... the only use-case for that (afaics) is to be able to view build warnings. Is there any way we could get Travis-CI to notify us about build warnings without actually marking the build as failed?